### PR TITLE
Add equals() algorithm to compare points and boxes

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by aprokop** [comment]
Why not `operator==`?

File: packages/Search/src/details/DTK_DetailsAlgorithms.hpp [hunk 1]
```diff
@@ -19,6 +19,24 @@ namespace DataTransferKit
...
+            return false;
+    return true;
+}
+
+KOKKOS_INLINE_FUNCTION
+bool equals( Box const &l, Box const &r )
+{
+    return equals( l.minCorner(), r.minCorner() ) &&
+           equals( l.maxCorner(), r.maxCorner() );
+    ;
```
Comment (Rombur) [inline packages/Search/src/details/DTK_DetailsAlgorithms.hpp]: extra `;`

**Comment by dalg24** [comment]
> Why not `operator==`?

I am following Boost.Geometry here.  Eventually I would like to be able to enable other models for Point and Box (e.g. directly reading `Kokkos::View<Coordinate **, DeviceType>` instead of converting to `Kokkos::View<Point *, DeviceType>`) via some traits/adapters.  This way the new type would not need to come with the `operator==()` defined.  Also it will be easier to grep this way if we do decide we want to change, for instance to introduce a variant that allows some tolerance in the comparison.

**Comment by Rombur** [comment]
OK to merge for me. What about you @aprokop ?

**Comment by aprokop** [comment]
I'm fine too.

**Pull request merged**